add_trailing_comma/_plugins/_with.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import ast
+from typing import Iterable
+
+from tokenize_rt import Offset
+from tokenize_rt import Token
+
+from add_trailing_comma._ast_helpers import ast_to_offset
+from add_trailing_comma._data import register
+from add_trailing_comma._data import State
+from add_trailing_comma._data import TokenFunc
+from add_trailing_comma._token_helpers import END_BRACES
+from add_trailing_comma._token_helpers import find_simple
+from add_trailing_comma._token_helpers import fix_brace
+from add_trailing_comma._token_helpers import START_BRACES
+
+
+def _fix_with(i: int, tokens: list[Token]) -> None:
+    paren_offset = 1
+    while tokens[i + paren_offset].name == 'UNIMPORTANT_WS':
+        paren_offset += 1
+    if tokens[i + paren_offset].src == '(':
+        multiline_withitems = False
+        first_brace = i + paren_offset
+        brace_stack = [first_brace]
+
+        for i in range(first_brace + 1, len(tokens)):
+            token = tokens[i]
+            if token.src in START_BRACES:
+                brace_stack.append(i)
+            elif token.src in END_BRACES:
+                brace_stack.pop()
+            if len(brace_stack) == 1 and token.name == 'NL':
+                multiline_withitems = True
```

Comment:
why the stack check here?

Reply:
My idea was basically to not rewrite parentheses based on internal line break
```
with (open(
    "a",
    some_other_really_long_parameter=True,
) as a, self.lock: pass
```
TBH this example isn't perfect as parenthesized expression will be broken the next line which would make the with statement have a new line in it at the top level.
Would love to know your thoughts on this.

Reply:
isn't the `:` check below going to cover that?

Reply:
and when you've got weird cases like that -- include them in tests!

Reply:
I made a typo, sorry!
```python
with (open(
    "a",
    some_other_really_long_parameter=True,
) as a, self.lock): pass```

Reply:
I would expect it to come out like this

```python
with (
    open(
        "a",
        some_other_really_long_parameter=True,
    ) as a, self.lock,
): pass
```

and you branch _does_ do that -- but it takes 2 passes to get there